gradle/docker-build.gradle

```diff
@@ -0,0 +1,55 @@
+// script to build/push docker images.
+// The script setup, opposite to most of other gradle plugins, is quite obfuscated,
+// hence requires dedicated `buildscript` block and special classes imports.
+// see https://github.com/bmuschko/gradle-docker-plugin#remote-api-plugin
+// http://bmuschko.com/blog/dockerized-spring-boot-app/
+// Also, we don't use custom createDockerfile task in favor of static `docker-build/Dockerfile`,
+// which could be reused, for instance, for local native docker builds.
+// This could be changed in the future if we really need custom runtime flexible Dockerfile managing
+
+buildscript {
+    repositories {
+        jcenter()
+    }
+    dependencies {
+        classpath "com.bmuschko:gradle-docker-plugin:$dockerPackageVersion"
+    }
+}
+
+apply plugin: com.bmuschko.gradle.docker.DockerRemoteApiPlugin
+
+apply from: "$rootDir/gradle/util.gradle"
+
+import com.bmuschko.gradle.docker.tasks.image.DockerBuildImage
+import com.bmuschko.gradle.docker.tasks.image.DockerPushImage
+
+// project specific settings:
+final dockerContextDir = file('.')
+final dockerfileFile = file('docker-build/Dockerfile')
+final dockerRepository = 'commercetoolsps/commercetools-paypal-plus-integration'
+final dockerUserProperty = 'docker.username'
+final dockerPassProperty = 'docker.password'
+
+task buildDockerImage(type: DockerBuildImage) {
+    dependsOn assemble
+
+    inputDir = dockerContextDir
+    dockerFile = dockerfileFile
+
+    tag = "$dockerRepository:$project.version"
```

Comment:
So now we don't use `latest` tag anymore and always use git hash?

Reply:
also the `"wip-${TRAVIS_BRANCH//\//-}"` tag if it's not master https://github.com/commercetools/commercetools-paypal-plus-integration/pull/124/files#diff-3f771c404ece392ae5c9f42234194c61L15

Reply:
this is not true any more: now only `dev-version-[commit]` or `git-tag` docker image tag is applied. 
We don't push `latest`, `master`, `travis-`, `wip-` or similar any more - they are confusing.

Reply:
ok, maybe. It's always good to notify you about it. Since it wasn't even mentioned in the description. In case you missed it.

Reply:
agree, will put it to release notes at least